File: src/foam/dao/ORBRequest.java
package foam.dao;

import foam.core.FObject;

/**
 * ORBRequest objects are used to pass method call arguments to ORBitalDAOs request brokers.
 */
public class ORBRequest{

  /** receiverObject can be the ID of the Object or a direct reference to it */
  private FObject receiverObject;
  private String receiverObjectID;

  private String methodName;
  private Object[] args;

  /** An ORBRequest can be constructed using a reference to the actual request receiver */
  public ORBRequest(FObject receiverObject, String methodName, Object[] args){
    this.receiverObject = receiverObject;
    this.methodName = methodName;
    this.args = args;
  }

  /** An ORBRequest can be constructed using the id of the request receiver */
  public ORBRequest(String receiverObjectID, String methodName, Object[] args){
    this.receiverObjectID = receiverObjectID;
    this.methodName = methodName;
    this.args = args;
  }

  public FObject getReceiverObject() {
    return receiverObject;
  }

  public String getReceiverObjectID() {
    return receiverObjectID;
  }

  public String getMethodName() {
    return methodName;
  }

  public Object[] getArgs() {
    return args;
  }

}

File: src/foam/dao/ORBRequest.js
/**
 * @license
 * Copyright 2019 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

foam.CLASS({
  package: 'foam.dao',
  name: 'ORBRequest',
  documentation : 'ORBRequest objects are used to pass method call arguments to ORBitalDAOs request brokers.',
  flags: ['java'],

  javaImports: [
    'foam.core.FObject'
  ],

  properties: [
    {
      name: 'receiverObject',
      class: 'FObjectProperty',
      documentation: 'receiverObject can be the ID of the Object or a direct reference to it.'
    },
    {
      name: 'receiverObjectID',
      class: 'String'
    },
    {
      name: 'methodName',
      class: 'String'
    },
    {
      name: 'args',
      class: 'Array'
    }
  ]

});
